specifications/api/swagger.yaml

```diff
@@ -790,6 +790,8 @@ x-stakePoolMetadata: &stakePoolMetadata
       type: string
       format: uri
       example: https://iohk.io
+    delisted:
+      type: boolean
```

Comment:
1) Booleans in API are generally a bad idea as they tend to bite us later and create undesirable breaking change. I'd highly recommend using a sum-type to capture the semantic of this field. 

2) This is not a metadata and as such, should not figure in this particular place. Let's move it one level higher with the rest of the pool's field.

Reply:
Next time I definitely want a complete swagger spec before I start working on the API.

Reply:
Please provide a suggestion for the change in OpenAPI spec.

Reply:
> This should be filled in by the story users/API consumers

No way. The API is defined by us, not by users (that's a recipe for disaster). In principle, it's your role a a pilot to define the API spec or challenge one defined in the acceptance criteria if any, and get it validated. 

> Please provide a suggestion for the change in OpenAPI spec.

I already did: 

```hs
data PoolFlag = NoPoolFlag | Delisted
```

Reply:
> No way. The API is defined by us, not by users (that's a recipe for disaster). In principle, it's your role a a pilot to define the API spec or challenge one defined in the acceptance criteria if any, and get it validated.

So far it has been that the API has been challenged several times in the course of the PR although most of it was written down in the ticket. I don't think that scales.

> I already did:

Yes and the discussion with Jonathan has yielded that we dropped this type.

Reply:
@KtorZ I'm trying to figure out the motivation for this sum type:

```hs
data PoolFlag = NoPoolFlag | Delisted
```

If we're just trying to indicate whether a pool is delisted, then `Delisted` does indeed convey this information.

However, `NoPoolFlag` doesn't seem to tell the user anything. Consider the serialized representation:

```json
{   "id": "pool15pvw2h7q43fkjt66nf5ezwn57ksxk3nqyds3yzk5a6jxss7t0d6"
,   "flag": "no-pool-flag"
}
```
If I look at the above representation, then I can't tell that this flag has anything to do with delisting.

Compare the above with:
```json
{   "id": "pool15pvw2h7q43fkjt66nf5ezwn57ksxk3nqyds3yzk5a6jxss7t0d6"
,   "delisted-status": "not-delisted"
}
```
The meaning is now immediately obvious.

However, if we're attempting to anticipate a future where there are many flags, then why not just provide a list:
```json
{   "id": "pool15pvw2h7q43fkjt66nf5ezwn57ksxk3nqyds3yzk5a6jxss7t0d6"
,   "flags": ["delisted"]
}
```
The "no pool flag" case is represented by the _empty list_.

This design also allows older clients to ignore flags that are introduced later on.

Reply:
Note that there's a difference between the Haskell data-type and how we serialize it into JSON. For example, a type like: `data PoolFlag = NoPoolFlag | Delisted` could very much be serialized as `[] | ["delisted"]` :). 

Although arguably, we could make the Haskell data-type also capture this and have a `flags :: [PoolFlag]` with `data PoolFlag = Delisted` and we can add more flags as we go.

Reply:
> Booleans in API are generally a bad idea as they tend to bite us later and create undesirable breaking change. I'd highly recommend using a sum-type to capture the semantic of this field.

Boolean is a sum type. I don't see why we'd want to avoid it. There's no boolean blindness here. In the API it's an object with key (string) and value (boolean).

> data PoolFlag = NoPoolFlag | Delisted

I don't understand the design of this type or what a flag is. It seems a little bit like preliminary optimization to me.

However, I don't have a strong opinion, but I already accepted a PR that removes this type. If you want to add it back, feel free to provide a PR.

> This is not a metadata and as such, should not figure in this particular place. Let's move it one level higher with the rest of the pool's field.

Please provide the swagger spec diff you'd like to see.

Reply:
> Boolean is a sum type

Thanks. May we not play on words here :) ? Boolean is a primitive sum types that we have no power to extend. Avoiding boolean in the API is not a premature optimization (it is not even an optimization) it makes future maintainability easier. 

> However, I don't have a strong opinion

I do :), so please, no Boolean.

> Please provide the swagger spec diff you'd like to see.

At this point I'll just do it myself, thanks.